modules/core/src/v2/coins/sol.ts

```diff
@@ -99,9 +106,59 @@ export class Sol extends BaseCoin {
     return Math.pow(10, this._staticsCoin.decimalPlaces);
   }
 
-  // TODO (https://bitgoinc.atlassian.net/browse/STLX-10202)
-  async verifyTransaction(params: VerifyTransactionOptions): Promise<boolean> {
-    throw new MethodNotImplementedError('verifyTransaction method not implemented');
+  async verifyTransaction(params: SolVerifyTransactionOptions): Promise<any> {
+    let totalAmount = 0;
+    const coinConfig = coins.get(this.getChain());
+    const {
+      txParams: txParams,
+      txPrebuild: txPrebuild,
+      memo: memo,
+      feePayer: feePayer,
+      blockhash: blockhash,
+      durableNonce: durableNonce,
+    } = params;
+    const transaction = new accountLib.Sol.Transaction(coinConfig);
+
+    if (!txPrebuild.txBase64) {
+      throw new Error('missing required tx prebuild property txBase64');
+    }
+
+    transaction.fromRawTransaction(txPrebuild.txBase64);
+    const explainedTx = transaction.explainTransaction();
+
+    if (!_.isEqual(explainedTx.outputs, txParams.recipients)) {
+      throw new Error('Tx outputs does not match with expected txParams recipients');
+    }
+
+    const transactionJson = transaction.toJson();
+    if (memo != explainedTx.memo) {
+      throw new Error('Tx memo does not match with expected txParams recipient memo');
+    }
+
+    if (explainedTx.outputs.length != txParams.recipients?.length) {
+      throw new Error('Tx outputs does not match with expected txParams recipient outputs');
+    }
```

Comment:
this is redundant.

there is an `isEqual` check on line 129, which should return false if the two arrays are of different lengths.